src/compose/resolve-block-map.ts

```diff
@@ -102,7 +102,27 @@ export function resolveBlockMap(
         : composeEmptyNode(ctx, offset, sep, null, valueProps, onError)
       offset = valueNode.range[2]
       const pair = new Pair(keyNode, valueNode)
-      if (ctx.options.keepSourceTokens) pair.srcToken = collItem
+      if (ctx.options.keepSourceTokens) {
+        pair.srcToken = collItem
+      }
+
+      // Check to see if the key tokens exist and have an indentation set.
+      // If so, we can compute the difference between them and preserve it
+      // if the preserveCollectionIndentation option is set.
+      const keyIndent: number | undefined =
+        key && 'indent' in key ? key.indent : undefined
+      const valueIndent: number | undefined =
+        value && 'indent' in value ? value.indent : undefined
```

Comment:
It would probably be clearer to skip defining these intermediate variables, and just use the conditionals in the subsequent if statement directly.